### PR TITLE
Sim: loadable module definitions

### PR DESCRIPTION
**Pull request by yamt**

### CODE REVIEW

File: boards/sim/sim/sim/configs/cxxtest/Make.defs [hunk 1]
```diff
@@ -88,10 +88,24 @@ CXXFLAGS = $(ARCHWARNINGSXX) $(ARCHOPTIMIZATION) \
...
+else
+  LDMODULEFLAGS += -T $(TOPDIR)/libs/libc/modlib/gnu-elf.ld
+endif
 
 # ELF module definitions
 
 CELFFLAGS = $(CFLAGS)
+# -fno-pic to avoid GOT relocations
+CELFFLAGS += -fno-pic
 CXXELFFLAGS = $(CXXFLAGS)
```
Comment (xiaoxiang781216) [inline boards/sim/sim/sim/configs/cxxtest/Make.defs]: need -fnop-pic too

File: boards/sim/sim/sim/configs/cxxtest/Make.defs [hunk 2]
```diff
@@ -128,6 +142,7 @@ ifeq ($(CONFIG_SIM_M32),y)
   LDLINKFLAGS += -melf_i386
   CCLINKFLAGS += -m32
   LDFLAGS += -m32
+  LDMODULEFLAGS += -melf_i386
```
Comment (xiaoxiang781216) [inline boards/sim/sim/sim/configs/cxxtest/Make.defs]: LDELFFLAGS need -melf_i386 too

File: boards/sim/sim/sim/configs/nsh2/Make.defs [hunk 3]
```diff
@@ -120,6 +134,7 @@ ifeq ($(CONFIG_SIM_M32),y)
   LDLINKFLAGS		+= -melf_i386
   CCLINKFLAGS		+= -m32
   LDFLAGS		+= -m32
+  LDMODULEFLAGS += -melf_i386
```
Comment (xiaoxiang781216) [inline boards/sim/sim/sim/configs/nsh2/Make.defs]: align with other line

File: boards/sim/sim/sim/configs/nsh2/Make.defs [hunk 4]
```diff
@@ -84,10 +84,24 @@ CXXFLAGS		= $(ARCHWARNINGSXX) $(ARCHOPTIMIZATION) \
...
+else
+  LDMODULEFLAGS += -T $(TOPDIR)/libs/libc/modlib/gnu-elf.ld
+endif
 
 # ELF module definitions
 
 CELFFLAGS = $(CFLAGS)
+# -fno-pic to avoid GOT relocations
+CELFFLAGS += -fno-pic
 CXXELFFLAGS = $(CXXFLAGS)
```
Comment (xiaoxiang781216) [inline boards/sim/sim/sim/configs/nsh2/Make.defs]: same comment as previous file

File: boards/sim/sim/sim/configs/cxxtest/Make.defs (hunk 1 above)
Comment (yamt) [reply boards/sim/sim/sim/configs/cxxtest/Make.defs]: oops, fixed

File: boards/sim/sim/sim/configs/cxxtest/Make.defs (hunk 2 above)
Comment (yamt) [reply boards/sim/sim/sim/configs/cxxtest/Make.defs]: done

File: boards/sim/sim/sim/configs/nsh2/Make.defs (hunk 3 above)
Comment (yamt) [reply boards/sim/sim/sim/configs/nsh2/Make.defs]: done

File: boards/sim/sim/sim/configs/nsh2/Make.defs (hunk 4 above)
Comment (yamt) [reply boards/sim/sim/sim/configs/nsh2/Make.defs]: done for all three files

**Pull request merged**